kie-spring-boot/kie-spring-boot-samples/jbpm-spring-boot-sample-basic/src/test/java/org/jbpm/springboot/samples/KafkaProxySampleTest.java

```diff
@@ -0,0 +1,189 @@
+import static org.apache.kafka.clients.producer.ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG;
+import static org.apache.kafka.clients.producer.ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG;
+
+import static org.jbpm.springboot.samples.KafkaFixture.KAFKA_PROCESS_ID;
+import static org.jbpm.springboot.samples.KafkaFixture.KAFKA_RESULT;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(Parameterized.class)
+@SpringBootTest(classes = {JBPMApplication.class, TestAutoConfiguration.class}, webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@TestPropertySource(locations="classpath:application-test.properties")
+@DirtiesContext(classMode= DirtiesContext.ClassMode.AFTER_CLASS)
+public class KafkaProxySampleTest extends KafkaBaseTest {
+
+    private static final Logger logger = LoggerFactory.getLogger(KafkaProxySampleTest.class);
+    
+    @ClassRule
+    public static final SpringClassRule scr = new SpringClassRule();
+ 
+    @Rule
+    public final SpringMethodRule smr = new SpringMethodRule();
+
+    @Autowired
+    private DeploymentService deploymentService;
+    
+    @Autowired
+    private ProcessService processService;
+    
+    @Autowired
+    CountDownLatchEventListener countDownLatchEventListener;
+    
+    protected static KafkaFixture kafkaFixture = new KafkaFixture();
+    
+    protected String deploymentId;
+    
+    protected String proxyBootstrap;
+
+    protected ToxiproxyContainer.ContainerProxy kafkaProxy;
+    
+    @BeforeClass
+    public static void generalSetup() {
+        kafkaFixture.generalSetup();
+    }
+
+    @Before
+    public void setup() throws IOException, InterruptedException {
+        toxiproxy.start();
+        kafkaProxy = toxiproxy.getProxy(kafka, 9093);
+        kafka.start();
+        kafkaFixture.createTopic(kafka);
+        proxyBootstrap = kafkaProxy.getContainerIpAddress()+":"+kafkaProxy.getProxyPort();
+        
+        System.setProperty(BOOTSTRAP_SERVERS_CONFIG, proxyBootstrap);
+        System.setProperty(CLIENT_ID_CONFIG, "test_jbpm");
+        System.setProperty(KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
+        System.setProperty(VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
+        
+        deploymentId = kafkaFixture.setup(deploymentService, strategy);
+        
+        countDownLatchEventListener.setVariable(KAFKA_RESULT);
+    }
+
+    @After
+    public void cleanup() {
+        kafka.stop();
+        toxiproxy.stop();
+        kafkaFixture.cleanup(deploymentService);
+    }
+    
+    @Rule
+    public Network network = Network.newNetwork();
+
+    @Rule
+    public KafkaContainer kafka = new KafkaContainer()
+                                        .withExposedPorts(9093)
+                                        .withNetwork(network);
+    
+    
+    @Rule
+    public ToxiproxyContainer toxiproxy  = new ToxiproxyContainer().withNetwork(network);
```

Comment:
I would move this up close to the rest of the rules, so it's easier to follow when looking at the code

Reply:
done